main.py

```diff
@@ -347,109 +353,249 @@ async def score(self, interaction: discord.Interaction, player: str):
 
         await interaction.followup.send(embed=embed)
 
-    async def breakdown(self, interaction: discord.Interaction, player: str):
+    async def breakdown(
+        self, interaction: discord.Interaction, player: Optional[str] = None
+    ):
         """Compute player score with complete source enumeration.
 
         Arguments:
             interaction: Discord Interaction from CommandTree.
-            player: Runescape username to break down clan score for.
+            (optional) player: Runescape username to break down clan score for.
         """
-        if not validate_player_name(player):
-            await interaction.response.send_message(
-                "FAILED_PRECONDITION: RSNs can only be 12 characters long."
-            )
+
+        await interaction.response.defer(thinking=True)
+
+        if player is None:
+            player = interaction.user.display_name
+
+        try:
+            member, player = validate_user_request(interaction, player)
+        except (ReferenceError, ValueError) as error:
+            await send_error_response(interaction, str(error))
             return
 
         logging.info(
             f"Handling '/breakdown player:{player}' on behalf of "
             f"{normalize_discord_string(interaction.user.display_name)}"
         )
-        await interaction.response.defer()
 
         try:
-            points_by_skill, points_by_activity = score_total(player)
-        except RuntimeError as e:
-            await interaction.followup.send(str(e))
+            skills, clues, raids, bosses = score_info(player)
+        except RuntimeError as error:
+            await send_error_response(interaction, str(error))
             return
 
+        activities = clues + raids + bosses
+
         skill_points = 0
-        for _, v in points_by_skill.items():
-            skill_points += v
+        for skill in skills:
+            skill_points += skill["points"]
 
         activity_points = 0
-        for _, v in points_by_activity.items():
-            activity_points += v
+        for activity in activities:
+            activity_points += activity["points"]
+
+        points_total = skill_points + activity_points
+        rank_name = get_rank_from_points(points_total)
+        rank_color = get_rank_color_from_points(points_total)
+        rank_icon = find_emoji(self._discord_client.emojis, rank_name)
+
+        rank_breakdown_embed = build_response_embed(
+            f"{rank_icon} {member.display_name} | Rank Ladder",
+            "The **Iron Forged** player rank ladder.",
+            rank_color,
+        )
+
+        for rank in RANKS:
```

Comment:
Since this is going to be calculated every call for every member, consider pre-calculating for every rank on startup and just store in dict <rand:rank_embed>